### PR TITLE
[TF FE] Support complex tensors for BiasAdd operation

### PR DESCRIPTION
**Pull request by Pranshu-S**
Extending support for BiasAdd operation to complex tensors.
Resolves Issue: #22945

### CODE REVIEW

File: src/frontends/tensorflow_common/src/op/bias_add.cpp
Comment (rkazants) [inline src/frontends/tensorflow_common/src/op/bias_add.cpp:69]: let us call `set_node_name` for `complex_add_res`. In this case you can simplify your code a bit, please take at the code below and check duplication:)

File: src/frontends/tensorflow_common/src/op/bias_add.cpp
Comment (rkazants) [inline src/frontends/tensorflow_common/src/op/bias_add.cpp:30]:
```suggestion
    auto complex_type_inputs = (complex_type_mark_value && complex_type_mark_bias) ? true : false;
    // validations prior to processing
    if (complex_type_inputs) {
```
Let us assume that the graph is valid and consistent provided by TF. So no need to check

**Comment by rkazants** [comment]
build_jenkins

**Comment by Pranshu-S** [comment]
Hey @rkazants 
Thanks for the suggestions. Will update the PR.

**Comment by rkazants** [comment]
> Hey @rkazants Thanks for the suggestions. Will update the PR. Also, could you help me out on the ci/jenkins and ie_tests_win_gpu_vs2019_release test failures - Not able to view the details.

Do not pay attention to failures in ie_tests_win_gpu_vs2019_release. It is sporadic issue:)

File: tests/layer_tests/tensorflow_tests/test_tf_BiasAdd.py
Comment (rkazants) [inline tests/layer_tests/tensorflow_tests/test_tf_BiasAdd.py:199]: please parameterize this test on `data_format` input that can be 'NHWC' and 'NCHW'. Example of parametrization: https://github.com/openvinotoolkit/openvino/blob/master/tests/layer_tests/tensorflow_tests/test_tf_StringLower.py#L40

File: tests/layer_tests/tensorflow_tests/test_tf_BiasAdd.py
Comment (rkazants) [inline tests/layer_tests/tensorflow_tests/test_tf_BiasAdd.py:201]:
```suggestion
    @pytest.mark.precommit_tf_fe
    @pytest.mark.nightly
```

File: src/frontends/tensorflow_common/src/op/bias_add.cpp
Comment (rkazants) [inline src/frontends/tensorflow_common/src/op/bias_add.cpp:53]: in case of complex type, value shape will contain aux dimension by which we should not unsqueeze bias. So `value_rank = value_shape.rank().get_length() - 1;`

File: src/frontends/tensorflow_common/src/op/bias_add.cpp
Comment (rkazants) [inline src/frontends/tensorflow_common/src/op/bias_add.cpp:55]:
```suggestion
                axes_unsqueeze.push_back(dim_ind);
```

**Comment by rkazants** [comment]
build_jenkins

**Review by rkazants — APPROVED** [review]

**Comment by Pranshu-S** [comment]
Hey @rkazants. I have raised a change in the PR. Do take a look

**Review by rkazants — APPROVED** [review]

**Comment by rkazants** [comment]
build_jenkins

**Comment by rkazants** [comment]
@Pranshu-S, let us wait for green CI. Once it passed, I will merge it.

Best regards,
Roman

**Comment by Pranshu-S** [comment]
Looking at the failures. Will raise a revision.
Thanks!

File: src/frontends/tensorflow_common/src/op/bias_add.cpp
Comment (Pranshu-S) [reply src/frontends/tensorflow_common/src/op/bias_add.cpp:53]: Hi @rkazants 
Regarding the recent change - i think the correct approach to handle the rank (in float transformed complex tensors) would be by skipping it in the right order. So instead I have reverted your recommended change and added an additional condition in the logic where we select the axis to squeeze. Do take alook

Comment (Pranshu-S) [reply src/frontends/tensorflow_common/src/op/bias_add.cpp:53]: However, do correct me if I am wrong, when we say a new dimension is added for supporting complex types does it get added such that:
Say initial dimension (of real or imag tensor) is [3,4]
Then the corresponding tensor would be 
- [2,3,4] (The assumption I have taken) 
- or [3,4,2]? 

Thanks!

Comment (Pranshu-S) [reply src/frontends/tensorflow_common/src/op/bias_add.cpp:53]: Hi @rkazants 
Bringing this up again as a reminder.

Thanks!

**Comment by mlukasze** [comment]
build_jenkins

**Comment by rkazants** [comment]
@Pranshu-S, please fix code-style failure and test failure. See CI logs

File: tests/layer_tests/tensorflow_tests/test_tf_BiasAdd.py
Comment (rkazants) [inline tests/layer_tests/tensorflow_tests/test_tf_BiasAdd.py:28]: revert changes in tests and simple add test for complex case

**Review by rkazants — CHANGES_REQUESTED** [review]
please revert changes in tests and simple add test for complex case

File: tests/layer_tests/tensorflow_tests/test_tf_BiasAdd.py
Comment (Pranshu-S) [reply tests/layer_tests/tensorflow_tests/test_tf_BiasAdd.py:28]: Hey Roman, my bad. 
I probably ran Clang for the entire local package which resulted in this interestingly. I'll revert.

**Comment by Pranshu-S** [comment]
Hey @rkazants 
Raised a revision with the changes. 

Thanks!

File: tests/layer_tests/tensorflow_tests/test_tf_BiasAdd.py
Comment (rkazants) [inline tests/layer_tests/tensorflow_tests/test_tf_BiasAdd.py:207]:
```suggestion
    @pytest.mark.precommit
```

**Review by rkazants — APPROVED** [review]

**Comment by rkazants** [comment]
build_jenkins

File: tests/layer_tests/tensorflow_tests/test_tf_BiasAdd.py
Comment (rkazants) [inline tests/layer_tests/tensorflow_tests/test_tf_BiasAdd.py:203]:
```suggestion
        dict(input_shape=[1, 1], bias_shape=[1], data_format="NHWC"),
        dict(input_shape=[3, 2, 7], bias_shape=[7], data_format="NHWC"),
        dict(input_shape=[3, 2, 7, 10], bias_shape=[2], data_format="NCHW"),
        dict(input_shape=[7, 6, 4, 5], bias_shape=[6], data_format="NCHW"),
```

File: tests/layer_tests/tensorflow_tests/test_tf_BiasAdd.py
Comment (rkazants) [inline tests/layer_tests/tensorflow_tests/test_tf_BiasAdd.py:207]:
```suggestion
    @pytest.mark.precommit
    @pytest.mark.nightly
```

**Comment by rkazants** [comment]
build_jenkins

File: tests/layer_tests/tensorflow_tests/test_tf_BiasAdd.py
Comment (rkazants) [inline tests/layer_tests/tensorflow_tests/test_tf_BiasAdd.py:178]:
```suggestion
    def create_complex_bias_add_net(self, input_shape, bias_shape, data_format, ir_version, use_legacy_frontend, output_type=tf.float32):
```

**Comment by rkazants** [comment]
build_jenkins

**Pull request merged**
